modules/runner/src/main/java/org/apache/ignite/internal/storage/LocalConfigurationStorage.java

```diff
@@ -49,47 +69,121 @@ public LocalConfigurationStorage(VaultManager vaultMgr) {
         this.vaultMgr = vaultMgr;
     }
 
-    /** Map to store values. */
-    private Map<String, Serializable> map = new ConcurrentHashMap<>();
-
     /** Change listeners. */
     private List<ConfigurationStorageListener> listeners = new CopyOnWriteArrayList<>();
 
     /** Storage version. */
     private AtomicLong version = new AtomicLong(0);
 
+    /** Start key in ragne for searching local configuration keys. */
+    private ByteArray localKeysStartRange = ByteArray.fromString(LOCAL_PREFIX + ".");
+
+    /** End key in range for searching local configuration keys. */
+    private ByteArray localKeysEndRange = ByteArray.fromString(LOCAL_PREFIX + (char)('.' + 1));
+
     /** {@inheritDoc} */
     @Override public synchronized Data readAll() throws StorageException {
-        return new Data(new HashMap<>(map), version.get(), 0);
+        Iterator<Entry> iter =
+            vaultMgr.range(localKeysStartRange, localKeysEndRange);
+
+        HashMap<String, Serializable> data = new HashMap<>();
+
+        while (iter.hasNext()) {
+            Entry val = iter.next();
+
+            data.put(val.key().toString().replaceFirst(LOCAL_PREFIX + ".", ""),
+                (Serializable)ByteUtils.fromBytes(val.value()));
+        }
+
+        return new Data(data, version.get());
```

Comment:
version is not persisted, right?

Reply:
yes, this will be needed when restart is developed